Gemfile

```diff
@@ -173,5 +173,6 @@ group :test do
   gem "vcr", "~> 6.0" # Record your test suite's HTTP interactions and replay them during future test runs for fast, deterministic, accurate tests
   gem "webdrivers", "~> 4.6" # Run Selenium tests more easily with install and updates for all supported webdrivers
   gem "webmock", "~> 3.13", require: false # WebMock allows stubbing HTTP requests and setting expectations on HTTP requests
+  gem "with_model", "~> 2.1.5" # Dynamically build a model within an RSpec context
```

Comment:
I had started to ad-hoc implement something like this on top of SQLite when writing the specs for this PR but thought that surely someone must have already done that. And indeed, there's a very complete gem that allows defining new models ad-hoc in tests: https://github.com/Casecommons/with_model

Reply:
Oh this is really cool!

Reply:
@joshpuetz It really is. I had something like this:

```ruby
class TestSettings < Settingss::Base
  establish_connection(adapter: "sqlite3", database: "tmp/test_settings.db")

  ...
end
```

That worked but at that point, I realized that I need to be able to define the schema for the backing table and started to wonder how to do this. I'm glad I found the gem because I got about 25% to implementing it before reason took over and I just googled. 😅